File: Java/BOJ/덱_10866.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.StringTokenizer;

public class µ¦_10866 {
	public static void main(String args[]) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int count = Integer.parseInt(br.readLine());
		StringTokenizer st;
		Deque deque = new ArrayDeque();
		StringBuilder sb = new StringBuilder();
		
		for(int i=0; i<count; i++) {
			st = new StringTokenizer(br.readLine(), " ");
			String type = st.nextToken();
			
			switch(type) {
			case "push_front":
				deque.addFirst(Integer.parseInt(st.nextToken()));
				break;
			case"push_back":
				deque.addLast(Integer.parseInt(st.nextToken()));
				break;
			case "pop_front":
				sb.append(deque.isEmpty() ? "-1" : deque.removeFirst()).append("\n");
				break;
			case "pop_back":
				sb.append(deque.isEmpty() ? "-1" : deque.removeLast()).append("\n");
				break;
			case "size":
				sb.append(deque.size()).append("\n");
				break;
			case "empty":
				sb.append(deque.isEmpty() ? "1" : "0").append("\n");
				break;
			case "front":
				sb.append(deque.isEmpty() ? "-1" : deque.peekFirst()).append("\n");
				break;
			case "back":
				sb.append(deque.isEmpty() ? "-1" : deque.peekLast()).append("\n");
				break;
			}
		}
		System.out.println(sb);
		br.close();
	}
}

File: Kotlin/BOJ/덱_10866.kt
package com.example.kotlincodingtest.baekjoon

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.ArrayDeque
import java.util.StringTokenizer

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val bw = BufferedWriter(OutputStreamWriter(System.`out`))
    val num = readLine().toInt()
    val deque = ArrayDeque<Int>()
    for(num in 0 until num) {
        val st = StringTokenizer(readLine(), " ")
        when(st.nextToken()) {
            "push_front" -> deque.addFirst(st.nextToken().toInt())
            "push_back" -> deque.addLast(st.nextToken().toInt())
            "pop_front" -> bw.write(if(deque.isEmpty()) "-1\n" else "${deque.pollFirst()}\n")
            "pop_back" -> bw.write(if(deque.isEmpty()) "-1\n" else "${deque.pollLast()}\n")
            "size" -> bw.write("${deque.size}\n")
            "empty" -> bw.write(if(deque.isEmpty()) "1\n" else "0\n")
            "front" -> bw.write(if(deque.isEmpty()) "-1\n" else "${deque.peekFirst()}\n")
            "back" -> bw.write(if(deque.isEmpty()) "-1\n" else "${deque.peekLast()}\n")
        }
    }
    bw.flush()
    bw.close()
    close()
}